spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    it 'returns a success response' do
+      user = User.create!(name: 'Name', posts_counter: 0)
+      get user_posts_path(user_id: user.id)
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('Here is a list of posts for a given user')
```

Comment:
- [ ] [OPTIONAL] Nice work making the requested changes, however, the project requirements states that your Spec file should include tests that ensures correct view template is rendered. I am making this optional because it was missed by the previous reviewer. 

            Example : expect(response).to render_template(:index) 

Kindly update your project accordingly to match this. :pray: